timemachine/fe/cif_writer.py

```diff
@@ -37,67 +34,84 @@ def convert_single_topology_mols(coords: np.ndarray, atom_map: AtomMapMixin) ->
 
         ```
         topol = app.PDBFile("1dfr.pdb").topology
         mol_a = Chem.MolFromMolBlock("...")
         mol_b = Chem.MolFromMolBlock("...")
 
-        writer = PDBWriter([topol, mol_a, mol_b], out.pdb) # writer header
+        writer = PDBxWriter([topol, mol_a, mol_b], out.pdb) # writer header
         writer.write_frame(coords) # coords must be in units of angstroms
         writer.close() # writes footer
         ```
 
         Parameters
         ----------
-        objs: list of either Chem.Mol or app.Topoolgy types
+        objs: list of either Chem.Mol or app.Topology types
             Molecules of interest
 
         out_filepath: str
-            Where we write out the PDBFile to
+            Where we write out the PDBxFile to
 
         """
 
         assert len(objs) > 0
 
-        rd_mols = []
-        # super jank
+        combined_topology = None
+
+        # see if an existing topology is present
         for obj in objs:
             if isinstance(obj, app.Topology):
-                with tempfile.NamedTemporaryFile(mode="w") as fp:
-                    # write
-                    PDBFile.writeHeader(obj, fp)
-                    PDBFile.writeModel(obj, np.zeros((obj.getNumAtoms(), 3)), fp, 0)
-                    PDBFile.writeFooter(obj, fp)
-                    fp.flush()
-                    # read
-                    rd_mols.append(Chem.MolFromPDBFile(fp.name, removeHs=False))
-
-            if isinstance(obj, Chem.Mol):
-                rd_mols.append(obj)
-
-        combined_mol = rd_mols[0]
-        for mol in rd_mols[1:]:
-            combined_mol = Chem.CombineMols(combined_mol, mol)
-
-        with tempfile.NamedTemporaryFile(mode="w") as fp:
-            # write
-            Chem.MolToPDBFile(combined_mol, fp.name)
-            fp.flush()
-            # read
-            combined_pdb = app.PDBFile(fp.name)
-            self.topology = combined_pdb.topology
+                assert combined_topology is None
+                combined_topology = obj
+
+        if combined_topology is None:
+            combined_topology = app.Topology()
+
+        for idx, obj in enumerate(objs):
+            if isinstance(obj, app.Topology):
+                pass
+            elif isinstance(obj, Chem.Mol):
+                mol = obj
+                new_chain = combined_topology.addChain()
+                new_residue = combined_topology.addResidue(name="LIG" + str(idx), chain=new_chain)
+                old_idx_to_new_atom_map = {}
+                for atom in mol.GetAtoms():
+                    name = atom.GetSymbol() + str(atom.GetIdx())
+                    element = app.element.Element.getByAtomicNumber(atom.GetAtomicNum())
+                    new_atom = combined_topology.addAtom(name, element, new_residue)
+                    old_idx = atom.GetIdx()
+                    old_idx_to_new_atom_map[old_idx] = new_atom
+
+                # (ytz): while we could get fancier, mmcif only has the covale bond type:
```

Comment:
Looks like you are referring to the code, but reads like a typo. Not sure if this matters

```suggestion
                # (ytz): while we could get fancier, mmcif only has the covalent bond type:
```

Reply:
Maybe just adding quotes to indicate that it is intentional? Anyways, just a nit since I read it as a typo

Reply:
done